### PR TITLE
Add ability to track data fields

### PR DESCRIPTION
**Pull request by chillbrodev**
This patch introduces a new function called trackWithDataFields.

In order to be backwards compatible with existing SDK users, I added a new function.

This gives sdk users the ability to add extra data to tracked events.

### CODE REVIEW

**Comment by chillbrodev** [comment]
I know there is another PR open for this but it seems dead. This is required by my project, so I figured I would add it in. Looking forward to feedback/integration. Thanks

File: lib/iterable_flutter.dart [hunk 1]
```diff
@@ -36,7 +36,13 @@ class IterableFlutter {
   }
 
   static Future<void> track(String event) async {
-    await _channel.invokeMethod('track', event);
+    await _channel.invokeMethod('track');
```
Comment (danielgomezrico) [inline lib/iterable_flutter.dart]: Any reason for deleting the event here? the track will be broken without it

File: lib/iterable_flutter.dart [hunk 2]
```diff
@@ -36,7 +36,13 @@ class IterableFlutter {
...
+  }
+
+  static Future<void> trackWithDataFields(
+      String event, Map<String, dynamic> dataFields) async {
+    await _channel.invokeMethod(
```
Comment (danielgomezrico) [inline lib/iterable_flutter.dart]: Why do you want to have a new method that adds the data fields there?

To me, it should be the same method but with optional params to keep it in sync with the original sdk, WDYT?

**Review by danielgomezrico — CHANGES_REQUESTED** [review]
Thanks for opening the PR

**Review by cdmunoz — APPROVED** [review]
Thks for your contribution!

With @danielgomezrico's suggestions implemented, it's ok with me

File: lib/iterable_flutter.dart (hunk 1 above)
Comment (chillbrodev) [reply lib/iterable_flutter.dart]: Oh shoot, this is my mistake. I didn't mean to remove it

File: lib/iterable_flutter.dart (hunk 2 above)
Comment (chillbrodev) [reply lib/iterable_flutter.dart]: I added a new method last night cuz that was my current thought pattern. I then later had this similar thought that we could use the same method with optional params. I can go ahead and make that change.

**Comment by chillbrodev** [comment]
Standby for changes

File: example/lib/main.dart [hunk 3]
```diff
@@ -137,7 +144,7 @@ class _MyAppState extends State<MyApp> {
             SizedBox(height: 20),
             ElevatedButton(
               onPressed: () {
-                  signOut();
+                signOut();
```
Comment (chillbrodev) [inline example/lib/main.dart:147]: @danielgomezrico @cdmunoz Is this formatting change okay? I can remove it if needed.

**Comment by chillbrodev** [comment]
@danielgomezrico @cdmunoz I have made the changes discussed above

**Review by danielgomezrico — APPROVED** [review]
🚀 Awesome, thanks for the effort

@chillbrodev can you check the unit tests that are failing?

**Comment by chillbrodev** [comment]
> 🚀 Awesome, thanks for the effort
> 
> @chillbrodev can you check the unit tests that are failing?

Oh sure thing, sorry its the week before the holiday break so been trying to cram in work. Missed the tests. Standby for a change

**Review by danielgomezrico — APPROVED** [review]
Sorry for taking too long but good job! deployed on https://github.com/la-haus/iterable-flutter/releases/tag/v0.5.8

**Pull request merged**